synapse/state/__init__.py

```diff
@@ -206,7 +207,9 @@ async def compute_state_after_events(
             holds the resolution of the states after the given event IDs.
         """
         logger.debug("calling resolve_state_groups from compute_state_after_events")
-        ret = await self.resolve_state_groups_for_events(room_id, event_ids)
+        ret = await self.resolve_state_groups_for_events(
+            room_id, event_ids, await_full_state
```

Comment:
Outside of `EventBuilder.build`, `compute_state_after_events` is only called in `RoomMemberHandler.update_membership_locked`.
Membership being a lot more tricky, I guess we don't want to use `await_full_state=False` in this case, so we can't just default to `False` here and avoid adding a parameter.

Reply:
Elsewhere, we try to dynamically calculate whether we should block, based on the state filter.

https://github.com/matrix-org/synapse/blob/3dfc4a08dc2e77178f2c2af68dc14b32da2d8b8f/synapse/storage/controllers/state.py#L259-L265

Maybe we can do the same thing here?